File: leetcode/cpp/sort-an-array-of-0s-1s-and-2s.cpp
#include <bits/stdc++.h>
using namespace std;
class Solution {
  public:
    void sort012(vector<int>& arr) {
        int i = 0, j = 0, l = arr.size();
        while(j<l){
            if(arr[j]==0){
                swap(arr[i], arr[j]);
                i++;
                j++;
            }else if(arr[j]==1){
                j++;
            }else{
                l--;
                swap(arr[j], arr[l]);
            }
        }
    }
};

File: GFG/java/sort-an-array-of-0s-1s-and-2s.java
class Solution{
    void swap(int[] arr, int i, int j){
        int temp = arr[i];
        arr[i] = arr[j];
        arr[j] = temp;
    }
    void sortArray(int[] arr){
        int n = arr.length;
        int i = 0, j = 0, k = n-1;
        while(j<k){
            if(arr[j]==0){
                swap(arr, i, j);
                i++;
                j++;
            }else if(arr[j]==2){
                swap(arr, j, k);
                k--;
            }else{
                j++;
            }
        }
    }
}